### ISSUE TITLE
Should the derivative of softmax be used in the first step of backpropagation?

### ISSUE THREAD

**Issue by Jiajun-Chu** [KirillShmilovich/MLP-Neural-Network-From-Scrath#4]
What a wonderful detailed MLP instance for a beginner like me! Thanks for your work! 👍 

I have a question about the following line of code : `delta_t = (output - batch_y)*sigmoid_prime(hidden_layers[-1])`

I am not sure if the derivative of softmax should be used in the first step of backpropagation. instead. Or is the sigmoid prime a simplified version in this instance?

Thank you in advance for your help!
